Meziantou.Polyfill.Tests/UnitTest1.cs

```diff
@@ -428,6 +428,20 @@ public void Enumerable_MinBy()
         Assert.Equal("a", new[] { "a", "ab", "b" }.MinBy(a => a.Length, Comparer<int>.Default));
     }
 
+    [Fact]
+    public void Enumerable_ToDictionary()
+    {
+        Dictionary<string, int> dict = new() { ["a"] = 1, ["b"] = 2 };
+
+        var copied = (Dictionary<string, int>) ((IEnumerable<KeyValuePair<string, int>>) dict).ToDictionary();
+        Assert.Equal(dict, copied);
+        Assert.Equal(dict.Comparer, copied.Comparer);
+
+        var caseInsensitive = (Dictionary<string, int>) ((IEnumerable<KeyValuePair<string, int>>) dict).ToDictionary(StringComparer.OrdinalIgnoreCase);
+        Assert.Equal(dict, caseInsensitive);
+        Assert.NotEqual(dict.Comparer, caseInsensitive.Comparer);
```

Comment:
I think you can assert the comparer. If the comparer is not correct, because of optimization in Dictionary for string keys, you can switch to `StringComparer.CurrentCulture` which should not be optimized.

```suggestion
        Assert.Equal(StringComparer.OrdinalIgnoreCase, caseInsensitive.Comparer);
```